### PR TITLE
Final `final` fixes; guards; cleanup.

### PR DESCRIPTION
**Pull request by pq**
* repaired mis-declaration of `FlutterJsonObject` fields as `final` (they’re not)
* NPE guards
* dead code cleanout

/cc @devoncarew

### CODE REVIEW

File: src/io/flutter/run/FlutterAppStateBase.java [hunk 1]
```diff
@@ -77,7 +77,7 @@ public FlutterAppStateBase(final @NotNull ExecutionEnvironment env) throws Execu
   @Override
   protected AnAction[] createActions(final ConsoleView console, final ProcessHandler processHandler, final Executor executor) {
     // These actions are effectively added only to the Run tool window. For Debug see DartCommandLineDebugProcess.registerAdditionalActions()
-    final List<AnAction> actions = new ArrayList(Arrays.asList(super.createActions(console, processHandler, executor)));
+    final ArrayList<AnAction> actions = new ArrayList<>(Arrays.asList(super.createActions(console, processHandler, executor)));
```
Comment (devoncarew) [inline src/io/flutter/run/FlutterAppStateBase.java]: `ArrayList` not `List`?

File: src/io/flutter/run/daemon/FlutterAppManager.java [hunk 2]
```diff
@@ -404,7 +404,7 @@ private void eventDeviceRemoved(@NotNull DeviceRemovedEvent removed, @NotNull Fl
       myService.getConnectedDevices().stream().filter(device -> device.deviceName().equals(removed.name) &&
                                                                 device.deviceId().equals(removed.id) &&
                                                                 device.platform().equals(removed.platform))
-        .forEach(device -> myService.removeConnectedDevice(device));
+        .forEach(myService::removeConnectedDevice);
```
Comment (devoncarew) [inline src/io/flutter/run/daemon/FlutterAppManager.java]: what crazy syntax

File: src/io/flutter/run/daemon/FlutterAppManager.java [hunk 3]
```diff
@@ -600,7 +600,7 @@ Event from(JsonElement element) {
   private static class LogMessageEvent extends Event {
     // "event":"daemon.eventLogMessage"
     @SuppressWarnings("unused") private String level;
-    @SuppressWarnings("unused") private final String message;
+    @SuppressWarnings("unused") private String message;
```
Comment (devoncarew) [inline src/io/flutter/run/daemon/FlutterAppManager.java]: oops, yeah, these are filled in via reflection

**Comment by devoncarew** [comment]
lgtm

File: src/io/flutter/run/FlutterAppStateBase.java (hunk 1 above)
Comment (pq) [inline src/io/flutter/run/FlutterAppStateBase.java]: Hmmm.  That was a mistake.  Fixed.

File: src/io/flutter/run/daemon/FlutterAppManager.java (hunk 2 above)
Comment (pq) [inline src/io/flutter/run/daemon/FlutterAppManager.java]: No kidding.  Trying to make piece with it best I can!

**Pull request merged**